### PR TITLE
Introduce API for P2P diagnostics

### PR DESCRIPTION
**Pull request by sh3ll3x3c**
- Added endpoint for local multi-address check
- Added endpoint for external peer multi-address check by using its `peerID`

### CODE REVIEW

File: src/api/v2/handlers/p2p.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+						"{message}. User provided peerID: {peer_id}. Observed: {observed}",
+					))
+				},
+				DialError::Denied { cause } => {
+					Error::bad_request_unknown(&format!("Connection denied. Reason: {cause}",))
+				},
+				DialError::Transport(_) => {
+					let message = "An error occurred while negotiating the transport protocol(s) on a connection";
+					Error::bad_request_unknown(&format!("{message}. Cause: {dial_error}"))
+				},
```
Comment (aterentic-ethernal) [inline src/api/v2/handlers/p2p.rs:105]: - Agreed, but we can check that earlier and treat this case as internal server error 
- Agrred, same as aboxe
- Also under the `_ =>` match
- Is end user is not responsible for this result, we should return 200OK and status
- Same as above
- Same as above
- Same as above

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## v1.9.0
 
 - Introduce `incomplete` block status for blocks without data transactions or for blocks lacking commitments due to a failure
+- Add first two endpoints of a new API for the P2P diagnostics
```
[the remark this answers is not in the document]
Comment (f4bricat0r) [reply CHANGELOG.md:6]: Yeah, new ones should be on top, an oversight.

[the remark this answers is not in the document]
Comment (sh3ll3x3c) [reply CHANGELOG.md:6]: Newest ones should def be on top, my bad.

File: src/api/v2/README.md [hunk 3]
```diff
@@ -243,6 +243,62 @@ If **app** mode is not active (or signing key is not configured and `data` is su
 HTTP/1.1 404 Not found
 ```
 
+# P2P Diagnostics API
+
+This API is intended to be used for P2P network observability and diagnostics.
+
+## **GET** `/v2/p2p/local/info`
```
[the remark this answers is not in the document]
Comment (sh3ll3x3c) [reply src/api/v2/README.md:250]: Maybe a redirect to the info endpoint...

**Review by aterentic-ethernal — APPROVED** [review]

**Pull request merged**